docs/DesignDocs/SGX_QuoteVerify_Integration.md

```diff
@@ -0,0 +1,522 @@
+
+```C
+/** Contains data that will allow an alternative quote verification policy. */
+typedef struct _sgx_ql_qv_supplemental_t
+{
+    uint32_t version;               ///< Supplemental data version
+    time_t earliest_issue_date;     ///< Earliest issue date of all the collateral (UTC)
+    time_t latest_issue_date;       ///< Latest issue date of all the collateral (UTC)
+    time_t earliest_expiration_date;///< Earliest expiration date of all the collateral (UTC)
+    time_t tcb_level_date_tag;      ///< The SGX TCB of the platform that generated the quote is not vulnerable
+                                    ///< to any Security Advisory with an SGX TCB impact released on or before this date
+                                    ///< See [Intel Security Center Advisories](https://www.intel.com/content/www/us/en/security-center/default.html)
+    uint32_t pck_crl_num;           ///< CRL Num from PCK Cert CRL
+    uint32_t root_ca_crl_num;       ///< CRL Num from Root CA CRL
+    uint32_t tcb_eval_ref_num;      ///< Lower number of the TCBInfo and QEIdentity
+    uint8_t root_key_id[48];        ///< ID of the collateral's root signer (hash of Root CA's public key SHA-384)
+    sgx_key_128bit_t pck_ppid;      ///< PPID from remote platform.  Can be used for platform ownership checks
+    sgx_cpu_svn_t tcb_cpusvn;       ///< CPUSVN of the remote platform's PCK Cert
+    sgx_isv_svn_t tcb_pce_isvsvn;   ///< PCE_ISVNSVN of the remote platform's PCK Cert
+    uint16_t pce_id;                ///< PCE_ID of the remote platform
+    uint8_t sgx_type;               ///< Indicate the type of memory protection available on the platform, it should be
+                                    ///< one of Standard (0) and Scalable (1)
+
+    // Multi-Package PCK cert related flags, they are only relevant to PCK Certificates issued by PCK Platform CA
+    uint8_t platform_instance_id[16];///< Value of Platform Instance ID, 16 bytes
+    pck_cert_flag_enum_t dynamic_platform; ///< Indicate whether a platform can be extended with additional packages
+                                           ///< via Package Add calls to SGX Registration Backend
+    pck_cert_flag_enum_t cached_keys;      ///< Indicate whether platform root keys are cached by SGX Registration Backend
+    pck_cert_flag_enum_t smt_enabled;      ///< Indicate whether a plat form has SMT (simultaneous multithreading) enabled
+
+} sgx_ql_qv_supplemental_t;
+```
+
+## Proposed Changes
+
+In order to smoothly switch quote verification from OE to SGX DCAP QVL/QvE, we
+propose to implement the change in 2 steps as below:
+- Step 1: Implement QVL-based quote verification and make it co-exist with the
+  existing implementation
+- Step 2: Remove existing implementation and makes the DCAP QVL a hard dependency
+
+### Step 1: Implement QVL-based quote verification and make it co-exist with the existing implementation
+If SGX DCAP QVL/QvE are not installed, quote verification will continue to use existing
```

Comment:
How do you uninstall quote verification on Windows?

Reply:
QVL/QvE are part of DCAP QL installer in Windows, so QVL/QvE would be uninstalled when user uninstall DCAP QL.

Reply:
> QVL/QvE are part of DCAP QL installer in Windows, so QVL/QvE would be uninstalled when user uninstall DCAP QL.

Is DCAP QL also for quote generation? If so, the user cannot uninstall it.

Reply:
Yes, DCAP QL is for quote generation.